src/db/DBInitializer.ts

```diff
@@ -72,18 +80,72 @@ class DBInitializer {
     }
 
+    private getMigrationGoal(): number {
+        // get all files in the migrations folder and get highest number from files like 5.sql
+        const files = fs.readdirSync(this.migrationsPath)
+        const migrationIds = files
+            .map((file): number | null => {
+                const match = file.match(/^(\d+)\.sql$/)
+                if (match) {
+                    return parseInt(match[1], 10)
+                }
+                return null
+            })
+            .filter((migrationId) => migrationId !== null) as number[]
+        return Math.max(...migrationIds)
+    }
+
+    private runMigration(migrationId: number): void {
+        const migrationFile = `${this.migrationsPath}/${migrationId}.sql`
+        const migrationData = fs.readFileSync(migrationFile, 'utf8')
+        this.runQueries(migrationData)
+    }
+
+    private async runMigrations(): Promise<void> {
+        console.log('Checking migrations ...')
+        //check if migrations table is present and fetch the latest migration_id
+        const migrationsTableExists = await this.checkTableExist('migrations')
+        let latestMigrationId = 0
+        if (migrationsTableExists) {
+            const result = (await this.pool.query(
+                'SELECT MAX(migration_id) as latestMigrationId FROM migrations'
+            )) as RowDataPacket[]
+            latestMigrationId = result[0][0].latestMigrationId || 0
+            console.log(`Latest migration id: ${latestMigrationId}`)
+        } else {
+            console.log('No migrations table found, init migration number 0')
+        }
+        const migrationIdGoal = this.getMigrationGoal()
+        console.log(`Migration id goal: ${migrationIdGoal}`)
+        for (
+            let migrationId = latestMigrationId + 1;
+            migrationId <= migrationIdGoal;
+            migrationId++
+        ) {
+            console.log(`Running migration number ${migrationId} ...`)
+            this.runMigration(migrationId)
+            console.log(`Migration ${migrationId} done`)
+        }
+        console.log('All migration work finished')
+    }
+
     public async init(): Promise<void> {
         await this.waitForDatabase()
+        // check if there is anything, otherwise create the tables
         const tablesExist = await this.checkTablesExist()
         if (!tablesExist) {
             console.log(
                 "MySQL tables don't exist in the specified db, creating them..."
             )
             await this.runQueries(this.schemaData)
             console.log('tables created')
+            // if there is already a schema, check if migrations are needed
+        } else {
+            await this.runMigrations()
         }
+        // queries that are always executed, e.g. to replace all views
         if (this.sqlToRun) {
-            console.log('Running sql statements ...')
+            console.log(`Running sql statements ...`)
```

Comment:
Does the formatter always change to backticks?

Reply:
nope, used a var but removed it again